webrender/src/prim_store.rs

```diff
@@ -58,15 +58,38 @@ pub struct PrimitiveCacheInfo {
     pub key: PrimitiveCacheKey,
 }
 
+#[derive(Debug)]
+pub enum PrimitiveClipInfo {
+    NoClip,
+    Complex(Rect<f32>, f32),
+    Region(ClipRegion),
+    GpuAddress(GpuStoreAddress),
+}
+
+impl PrimitiveClipInfo {
+    pub fn is_clipped(&self) -> bool {
```

Comment:
You are right, this part doesn't seem right. If it's called after after the `prepare_*` body (and I think it does), the only variant here would be `GpuAddress`. Will be fixed in a minute.